Lib/logging/config.py

```diff
@@ -361,7 +361,7 @@ class BaseConfigurator(object):
 
     WORD_PATTERN = re.compile(r'^\s*(\w+)\s*')
     DOT_PATTERN = re.compile(r'^\.\s*(\w+)\s*')
-    INDEX_PATTERN = re.compile(r'^\[\s*(\w+)\s*\]\s*')
+    INDEX_PATTERN = re.compile(r'^\[(.*?)\]\s*')
```

Comment:
I wonder if we want to allow something like `nested[]]` where the inner `]` is interpreted as a key

Reply:
I did not find a lot of clarity in the documentation around the charset allowed. However, I don't see a reason why not.
Let's wait for a few more people to review.

In case you are aware where the specs are, please point me to the same.

Reply:
I haven't found anything besides what is linked in the issue, I guess it depends on how sophisticated we want it to be..

Reply:
Just seeking a clarification here. 

If we have a key `abc ] def` and we try to use it in the cfg notation as follows:

```
cfg://adict[abc ] def
```
Shouldn't it technically evaluate as a key `abc `?
Am I missing something here?

Reply:
here is a sample run to explain what I suggested

```
import logging
import logging.handlers
import logging.config

sample = {"a" : 1, "a]b" : 4, "a][b" : 12, a" : 3, "a b" : 5}
config = {'adict' : sample}

bc = logging.config.BaseConfigurator(config)

>>> bc.convert('cfg://adict[a]')
1
>>> 

>>> bc.convert('cfg://adict[ a]')
1

>>> bc.convert('cfg://adict[a b]')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/peeyush/.pyenv/versions/3.11.1/lib/python3.11/logging/config.py", line 461, in convert
    value = converter(suffix)
            ^^^^^^^^^^^^^^^^^
  File "/Users/peeyush/.pyenv/versions/3.11.1/lib/python3.11/logging/config.py", line 431, in cfg_convert
    raise ValueError('Unable to convert '
ValueError: Unable to convert 'adict[a b]' at '[a b]'


>>> bc.convert('cfg://adict[a][b]')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/peeyush/.pyenv/versions/3.11.1/lib/python3.11/logging/config.py", line 461, in convert
    value = converter(suffix)
            ^^^^^^^^^^^^^^^^^
  File "/Users/peeyush/.pyenv/versions/3.11.1/lib/python3.11/logging/config.py", line 421, in cfg_convert
    d = d[idx]
        ~^^^^^
TypeError: 'int' object is not subscriptable


>>> bc.convert('cfg://adict[a]b]')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/peeyush/.pyenv/versions/3.11.1/lib/python3.11/logging/config.py", line 461, in convert
    value = converter(suffix)
            ^^^^^^^^^^^^^^^^^
  File "/Users/peeyush/.pyenv/versions/3.11.1/lib/python3.11/logging/config.py", line 431, in cfg_convert
    raise ValueError('Unable to convert '
ValueError: Unable to convert 'adict[a]b]' at 'b]'

```

Reply:
Makes sense to me.

The document reads

```
The latter form only needs to be used if the key contains spaces or non-alphanumeric characters.
```

So special characters are indeed allowed. 
I think the regex pattern which is configured right now is okay.

It will match for the very first `]` as expected.
The user would intuitively understand this constraint.

However, we can improve the [documentation](https://raw.githubusercontent.com/python/cpython/main/Doc/library/logging.config.rst) as well. 

We can add - 

```
Please note that we cannot use `[` and `]` in the key.
```

Reply:
> We can add

Sure, feel free to do that in this PR as well.